### PR TITLE
workaround map downlodaing issue on linux

### PR DESCRIPTION
**Pull request by betrok**
There are many maps in the vault, where the case of the archive name (always lowercase?) does not coincide with name of the directory in it. This leads to #1442.

Here is a workaround for case sensitive filesystems.

Maybe repacking all the maps would be a cleaner solution...

### CODE REVIEW

File: src/main/java/com/faforever/client/map/MapService.java [hunk 1]
```diff
@@ -412,6 +413,21 @@ public Path getPathForMap(String technicalName) {
...
 
+  public Path getPathForMapInsensitive(String approxName) {
+    String normalizedName = approxName.toLowerCase();
+    try (DirectoryStream<Path> stream = Files.newDirectoryStream(customMapsDirectory)) {
+      for (Path entry : stream) {
+        if (entry.getFileName().toString().toLowerCase().equals(normalizedName)) {
+          return entry;
+        }
+      }
+    } catch (IOException e) {
```
Comment (1-alex98) [inline src/main/java/com/faforever/client/map/MapService.java]: lets simply not catch it at all, returning nul only delays the issue and probably results in a NPE, just use 

```suggestion
noCatch(()->{
     //Your code
});
```

File: src/main/java/com/faforever/client/map/MapService.java [hunk 2]
```diff
@@ -485,7 +501,7 @@ public void evictCache() {
     }
 
     return taskService.submitTask(task).getFuture()
-        .thenAccept(aVoid -> noCatch(() -> addSkirmishMap(getPathForMap(folderName))));
```
Comment (1-alex98) [inline src/main/java/com/faforever/client/map/MapService.java]: What about the old method ? Are those all places where this issue exists

**Review by 1-alex98 — CHANGES_REQUESTED** [review]

**Comment by 1-alex98** [comment]
I would be happy if u join our chat service Zulip, send me your email and I will invite u.
alexander.von.trostorff@gmail.com

File: src/main/java/com/faforever/client/map/MapService.java (hunk 2 above)
Comment (betrok) [reply src/main/java/com/faforever/client/map/MapService.java]: At first glance we know exact name of the directory everywhere else.

File: src/main/java/com/faforever/client/map/MapService.java (hunk 1 above)
Comment (betrok) [reply src/main/java/com/faforever/client/map/MapService.java]: Some code actually uses null as a missing map indicator, hm...

**Comment by betrok** [comment]
My public email should be available in the github profile. It can be used for an invite if necessary.

File: src/main/java/com/faforever/client/map/MapService.java (hunk 1 above)
Comment (1-alex98) [reply src/main/java/com/faforever/client/map/MapService.java]: ok then

**Comment by betrok** [comment]
I use client build with these changes for a week already and it seems to work fine enough.
Can someone take a look and maybe merge it?

**Review by 1-alex98 — APPROVED** [review]

**Comment by 1-alex98** [comment]
tested on Windows works

**Pull request merged**